### PR TITLE
Tooling improvements

### PR DESCRIPTION
**Pull request by feliwir**
- [x] Add an image view to big editor
- [x] Add writeable BigArchiveEntryStreams
- [x] Allow creation & deletion of entries
- [x] Incorporate import & delete features into the big editor

### CODE REVIEW

File: src/OpenSage.FileFormats.Big/BigArchive.cs [hunk 1]
```diff
@@ -118,6 +134,136 @@ public BigArchiveEntry GetEntry(string entryName)
...
+            return contentSize;
+        }
+
+        private int CalculateTableSize()
+        {
+            int tableSize = 0;
+            foreach (var entry in _entries)
+            {
+                // Each entry has 4 bytes for the offset + 4 for size
+                tableSize += 8;
```
Comment (Tarcontar) [inline src/OpenSage.FileFormats.Big/BigArchive.cs:161]: maybe use a const ENTRY_SIZE here, that would also make the comment above obsolete

File: src/OpenSage.FileFormats.Big/BigArchiveEntry.cs [hunk 2]
```diff
@@ -41,5 +75,67 @@ public Stream Open()
...
+                    return OpenInUpdateMode();
+            }
+        }
+
+        internal void CloseStreams()
+        {
+            // if the user left the stream open, close the underlying stream for them
+            if (OutstandingWriteStream != null)
+            {
+                OutstandingWriteStream.Dispose();
```
Comment (Skleni) [inline src/OpenSage.FileFormats.Big/BigArchiveEntry.cs:137]: Should you set it to null here too?

Comment (Skleni) [reply src/OpenSage.FileFormats.Big/BigArchiveEntry.cs:137]: And you could use `OutstandingWriteStream?.Dispose()`

File: src/OpenSage.FileFormats/BinaryWriterExtensions.cs [hunk 3]
```diff
@@ -286,6 +286,24 @@ public static void WriteFourCc(this BinaryWriter writer, string fourCc, bool big
             }
         }
 
+        public static void WriteNullterminatedString(this BinaryWriter writer, in string content)
```
Comment (Skleni) [inline src/OpenSage.FileFormats/BinaryWriterExtensions.cs:289]: Should probably be WriteNull*T*erminatedString

File: src/OpenSage.FileFormats.Big/BigArchive.cs [hunk 4]
```diff
@@ -118,6 +134,136 @@ public BigArchiveEntry GetEntry(string entryName)
...
+                {
+                    if (entry.OutstandingWriteStream != null)
+                    {
+                        entry.OutstandingWriteStream.WriteTo(content);
+                    }
+                    else
+                    {
+                        using (var stream = entry.Open())
+                        {
+                            // Each entry has 4 bytes for the offset + 4 for size
```
Comment (Tarcontar) [inline src/OpenSage.FileFormats.Big/BigArchive.cs:228]: obsolete comment

**Pull request merged**